# Explorando Dados de Vendas de Carros no Ebay
 Este é um projeto guiado feito a partir de um desafio do site "Dataquest".
Aqui possuímos um dataset com 371528 e 20 linhas com dados de carros usados do eBay Kleinanzeigen. Em seguida, temos um pequeno dicionário para explicar sobre cada coluna. 
- dateCrawled : Se refere a quando este anúncio foi feito pela primeira vez
- name : Nome do carro.
* seller : Se o vendedor é particular ou revendedor.
* offerType : O tipo de listagem
* price : O preço do anúncio de venda do carro.
* abtest : Se a lista está incluída em um teste A / B.
* vehicleType : O tipo de veículo.
* yearOfRegistration : O ano em que o carro foi registrado pela primeira vez.
* gearbox : O tipo de transmissão.
* powerPS : A potência do carro em 'cavalos'.
* model : O nome do modelo do carro.
* kilometer : Quantos quilômetros o carro percorreu.
* monthOfRegistration : O mês em que o carro foi registrado pela primeira vez.
* fuelType : Que tipo de combustível o carro usa.
* brand : A marca do carro.
* notRepairedDamage : Se o carro apresentar um dano que ainda não foi reparado.
* dateCreated : A data em que a lista do eBay foi criada.
* nrOfPictures : O número de fotos no anúncio.
* postalCode : O código postal da localização do veículo.
* lastSeenOnline : Quando o rastreador viu este anunciante online pela última vez.

In [1]:
# Importando as bibliotecas necessárias para este projeto
import pandas as pd
import numpy as np

# Importando o dataset
autos = pd.read_csv('C:/Users/andre/Documents/my_datasets/autos.csv', encoding='Latin-1')

In [2]:
autos

dateCrawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        seller offerType  price   abtest vehicleType  yearOfRegistration  \
0       privat   Angebot    480     test         NaN                1993   
1       privat   Angebot  18300     test       coupe                2011   
2       privat   Angebot   9800     test         suv                2004   
3       privat   Angebot   1500     test  kleinwagen                2001   
4       privat   Angebot   3600     test  kleinwagen                2008   
...        ...       ...    ...      ...         ...                 ...   
371523  privat   Angebot   2200     test         NaN                2005   
371524  privat   Angebot   1199     test      cabrio                2000   
371525  privat   Angebot   9200     test         bus                1996   
371526  privat   Angebot   3400     test       kombi                2002   
371527  privat   Angebot  28990  control   limousine                2013   

          gearbox  powerPS        model  kilometer  monthOfRegistration  \
0         manuell        0         golf     150000                    0   
1         manuell      190          NaN     125000                    5   
2       automatik      163        grand     125000                    8   
3         manuell       75         golf     150000                    6   
4         manuell       69        fabia      90000                    7   
...           ...      ...          ...        ...                  ...   
371523        NaN        0          NaN      20000                    1   
371524  automatik      101       fortwo     125000                    3   
371525    manuell      102  transporter     150000                    3   
371526    manuell      100         golf     150000                    6   
371527    manuell      320      m_reihe      50000                    8   

       fuelType           brand notRepairedDamage          dateCreated  \
0        benzin      volkswagen               NaN  2016-03-24 00:00:00   
1        diesel            audi                ja  2016-03-24 00:00:00   
2        diesel            jeep               NaN  2016-03-14 00:00:00   
3        benzin      volkswagen              nein  2016-03-17 00:00:00   
4        diesel           skoda              nein  2016-03-31 00:00:00   
...         ...             ...               ...                  ...   
371523      NaN  sonstige_autos               NaN  2016-03-14 00:00:00   
371524   benzin           smart              nein  2016-03-05 00:00:00   
371525   diesel      volkswagen              nein  2016-03-19 00:00:00   
371526   diesel      volkswagen               NaN  2016-03-20 00:00:00   
371527   benzin             bmw              nein  2016-03-07 00:00:00   

        nrOfPictures  postalCode             lastSeen  
0                  0       70435  2016-04-07 03:16:57  
1                  0       66954  2016-04-07 01:46:50  
2                  0       90480  2016-04-05 12:47:46  
3                  0       91074  2016-03-17 17:40:17  
4                  0       60437  2016-04-06 10:17:21  
...              ...         ...                  ..

In [3]:
# Nós mostra informações sobre o dataset
autos.info()
autos.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
dateCrawled            371528 non-null object
name                   371528 non-null object
seller                 371528 non-null object
offerType              371528 non-null object
price                  371528 non-null int64
abtest                 371528 non-null object
vehicleType            333659 non-null object
yearOfRegistration     371528 non-null int64
gearbox                351319 non-null object
powerPS                371528 non-null int64
model                  351044 non-null object
kilometer              371528 non-null int64
monthOfRegistration    371528 non-null int64
fuelType               338142 non-null object
brand                  371528 non-null object
notRepairedDamage      299468 non-null object
dateCreated            371528 non-null object
nrOfPictures           371528 non-null int64
postalCode             371528 non-null int64
lastSeen              

price  yearOfRegistration        powerPS      kilometer  \
count  3.715280e+05       371528.000000  371528.000000  371528.000000   
mean   1.729514e+04         2004.577997     115.549477  125618.688228   
std    3.587954e+06           92.866598     192.139578   40112.337051   
min    0.000000e+00         1000.000000       0.000000    5000.000000   
25%    1.150000e+03         1999.000000      70.000000  125000.000000   
50%    2.950000e+03         2003.000000     105.000000  150000.000000   
75%    7.200000e+03         2008.000000     150.000000  150000.000000   
max    2.147484e+09         9999.000000   20000.000000  150000.000000   

       monthOfRegistration  nrOfPictures    postalCode  
count        371528.000000      371528.0  371528.00000  
mean              5.734445           0.0   50820.66764  
std               3.712412           0.0   25799.08247  
min               0.000000           0.0    1067.00000  
25%               3.000000           0.0   30459.00000  
50%               6.000000           0.0   49610.00000  
75%               9.000000           0.0   71546.00000  
max              12.000000           0.0   99998.00000

Algumas das coisas que podemos observar:
- O conjunto de dados contém 20 colunas, a maioria das quais são strings.
- Algumas colunas têm valores nulos, mas nenhuma tem mais do que ~ 20% dos valores nulos.
- Os nomes das colunas usam camelcase em vez do snakecase preferido do Python , o que significa que não podemos simplesmente substituir espaços por sublinhados.

In [4]:
# Imprime uma matriz com o nome das colunas do dataset
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [5]:
# Cria uma lista com os nomes que iremos usar nas colunas
nome_colunas = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'kilometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'number_pictures', 'postal_code',
       'last_seen']

# Substituímos os nomes antigos
autos.columns = nome_colunas

# Agora podemos ver os novos nomes das colunas já em uso
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price abtest vehicle_type  registration_year    gearbox  power_ps  model  \
0    480   test          NaN               1993    manuell         0   golf   
1  18300   test        coupe               2011    manuell       190    NaN   
2   9800   test          suv               2004  automatik       163  grand   
3   1500   test   kleinwagen               2001    manuell        75   golf   
4   3600   test   kleinwagen               2008    manuell        69  fabia   

   kilometer  registration_month fuel_type       brand unrepaired_damage  \
0     150000                   0    benzin  volkswagen               NaN   
1     125000                   5    diesel        audi                ja   
2     125000                   8    diesel        jeep               NaN   
3     150000                   6    benzin  volkswagen              nein   
4      90000                   7    diesel       skoda              nein   

            ad_created  number_pictures  postal_code            last_seen  
0  2016-03-24 00:00:00                0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00                0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00                0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00                0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00                0        60437  2016-04-06 10:17:21

## Corrigindo os nomes das colunas
<p> Substituí os nomes das colunas do dataset por nomes que indique mais facilmente qual é o conteúdo daquelas colunas. Para isto utilizamos o estilo "snake case" em que cada espaço é substituído por um caractere sublinhado (_) e a primeira letra de cada palavra escrita em minúsculas. </p>

In [6]:
autos.describe(include='all')

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

        abtest vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  number_pictures   postal_code  \
count             299468               371528         371528.0  371528.00000   
unique                 2                  114              NaN           NaN   
top                 nein  2016-04-03 00:00:00              NaN           NaN   
freq              263182                14450              NaN           NaN   
mean                 NaN                  NaN              0.0   50820.66764   
std                  NaN                  NaN              0.0   25799.08247   
min                  NaN                  NaN              0.0    1067.00000   
25%                  NaN                  NaN              0.0   30459.00000   
50%                  NaN                  NaN              0.0   49610.00000   
75%                  NaN                  NaN              0.0   71546.00000   
max                  NaN                  NaN              0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max      

## Analisando o preenchimento das informações nas colunas
<p> Vamos analisar os valores presentes nas colunas, para facilitar a análise deixarei exposto apenas as colunas onde notei inconsistências. </p>

In [7]:
autos['name'].value_counts()

Ford_Fiesta                                          657
BMW_318i                                             627
Opel_Corsa                                           622
Volkswagen_Golf_1.4                                  603
BMW_316i                                             523
                                                    ... 
BMW_Compact_Police_Car                                 1
Golf_V_8_Fach_bereift                                  1
Seat_Leon_2.0_TDI_DPF_FR_weiss                         1
Suche_Van/Kombi                                        1
Golf_1_Cabrio_Etienne_Aigner._Sehr_viel_Zubehoer.      1
Name: name, Length: 233531, dtype: int64

In [8]:
autos['price'].value_counts()

0         10778
500        5670
1500       5394
1000       4649
1200       4594
          ...  
23456         1
171000        1
21830         1
13485         1
8188          1
Name: price, Length: 5597, dtype: int64

In [9]:
autos['registration_year'].value_counts()

2000    24551
1999    22767
2005    22316
2006    20230
2001    20218
        ...  
1919        1
1915        1
7800        1
1400        1
8455        1
Name: registration_year, Length: 155, dtype: int64

In [10]:
autos['power_ps'].value_counts()

0        40820
75       24035
60       15907
150      15442
140      13585
         ...  
5815         1
16051        1
1360         1
5809         1
1398         1
Name: power_ps, Length: 794, dtype: int64

In [11]:
autos['registration_month'].value_counts()

0     37675
3     36170
6     33167
4     30918
5     30631
7     28958
10    27337
11    25489
12    25380
9     25074
1     24561
8     23765
2     22403
Name: registration_month, dtype: int64

In [12]:
autos['number_pictures'].value_counts()

0    371528
Name: number_pictures, dtype: int64

## Analisando o conteúdo das colunas
Vamos agora analisar o que havia de conteúdos estranhos:
- A coluna "name" possui entradas muito bagunçadas e com caracteres estranhos para descrever o nome dos carros, suas entradas não parecem possuir um padrão claro e contém caracteres especiais ou de outros idiomas, apesar disso quando pesquiso seus nomes no google sou enviada rapidamente para modelos de carros.
- A coluna "registration_year" possuí algumas entradas muito estranhas como: 7800, 1400, 8455. Não sei exatamente o que aconteceu aqui, mas essas entradas não deveriam ser possíveis porque não há como nenhum desses carros ter sido registrado em tempos tão longínquos. 
- A coluna "power_ps" possui muitos valores estranhos, há muitas entradas com o valor "0" que pode ser atribuído a esse valor ter sido colocado aleatoriamente pois não possuíam essa informação, mas também há as entradas com valores muitos irreais como: 5815, 16051, 1360, 5809, 1398.
- As colunas "registration_month" e "price" possuem muitos valores com 0 que pode ser atribuído a esse valor ter sido colocado aleatoriamente pois não possuíam essa informação.
- A coluna "number_pictures" está vazia.

In [25]:
# Converte os valores pro tipo int
autos['price'] = autos['price'].astype(int)

# Fornece dados sobre a coluna
autos['price'].describe()

count    356717.000000
mean       5904.910097
std       11911.501944
min           0.000000
25%        1150.000000
50%        2999.000000
75%        7400.000000
max      999999.000000
Name: price, dtype: float64

In [26]:
# Deixa no nosso dataset apenas os valores entre os mencionados
autos = autos[autos['price'].between(0,999999)]
autos['price'].value_counts()

0        10014
500       5468
1500      5093
1000      4366
1200      4332
         ...  
1656         1
11389        1
7291         1
17520        1
8188         1
Name: price, Length: 5530, dtype: int64

<p> Aqui deixamos apenas as linhas com preço entre 0 e 999999 </p>

In [15]:
# Retira a coluna 'number_pictures' do nosso dataset limpo
autos = autos.drop(columns=['number_pictures'])
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price abtest vehicle_type  registration_year    gearbox  power_ps  model  \
0    480   test          NaN               1993    manuell         0   golf   
1  18300   test        coupe               2011    manuell       190    NaN   
2   9800   test          suv               2004  automatik       163  grand   
3   1500   test   kleinwagen               2001    manuell        75   golf   
4   3600   test   kleinwagen               2008    manuell        69  fabia   

   kilometer  registration_month fuel_type       brand unrepaired_damage  \
0     150000                   0    benzin  volkswagen               NaN   
1     125000                   5    diesel        audi                ja   
2     125000                   8    diesel        jeep               NaN   
3     150000                   6    benzin  volkswagen              nein   
4      90000                   7    diesel       skoda              nein   

            ad_created  postal_code            last_seen  
0  2016-03-24 00:00:00        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00        60437  2016-04-06 10:17:21

<p> Como a coluna "number_pictures" estava apenas com valores "0" a retirei. </p> 

### Trabalhando com datas
<p> Possuímos 5 colunas com informações relacionadas a data no nosso dataset; as colunas 'date_crawled' e 'last_seen' foram adicionadas pelo rastreador, enquanto que as colunas 'ad_created', regristration_month e regristration_year foram adicionadas pelo website. </p>
<p> Agora, os date_crawled, last_seene ad_createdcolunas são todos identificados como valores de cadeia por pandas. Como essas três colunas são representadas como strings, precisamos converter os dados em uma representação numérica para que possamos entendê-los quantitativamente. </p>

In [16]:
autos['date_crawled'].value_counts(normalize=True, dropna=False)

2016-03-24 14:49:47    0.000019
2016-03-26 22:57:31    0.000016
2016-03-19 21:49:56    0.000016
2016-04-01 15:56:46    0.000013
2016-03-16 13:47:44    0.000013
                         ...   
2016-03-16 17:46:30    0.000003
2016-03-26 15:47:03    0.000003
2016-04-02 13:38:22    0.000003
2016-03-11 14:56:04    0.000003
2016-03-14 12:57:32    0.000003
Name: date_crawled, Length: 280468, dtype: float64

In [17]:
autos['ad_created'].value_counts(normalize=True, dropna=False)

2016-04-03 00:00:00    0.038887
2016-04-04 00:00:00    0.037743
2016-03-20 00:00:00    0.036464
2016-03-12 00:00:00    0.036012
2016-03-21 00:00:00    0.035810
                         ...   
2014-03-10 00:00:00    0.000003
2015-11-08 00:00:00    0.000003
2016-01-06 00:00:00    0.000003
2015-12-27 00:00:00    0.000003
2016-01-08 00:00:00    0.000003
Name: ad_created, Length: 114, dtype: float64

In [18]:
autos['last_seen'].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05 14:15:08    0.000003
2016-03-05 14:15:16    0.000003
2016-03-05 14:15:39    0.000003
2016-03-05 14:18:30    0.000003
2016-03-05 14:25:59    0.000003
                         ...   
2016-04-07 14:58:47    0.000024
2016-04-07 14:58:48    0.000024
2016-04-07 14:58:49    0.000030
2016-04-07 14:58:50    0.000030
2016-04-07 14:58:51    0.000003
Name: last_seen, Length: 182774, dtype: float64

In [19]:
autos['registration_year'].describe()

count    371458.00000
mean       2004.57946
std          92.87496
min        1000.00000
25%        1999.00000
50%        2003.00000
75%        2008.00000
max        9999.00000
Name: registration_year, dtype: float64

<p> Possuímos alguns valores muito estranhos nessa coluna 'registration_year', o valor mínimo se encontra muitos anos antes dos carros serem inventados e os valores máximos se encontram anos após a coleta dos dados. Uma pesquisa rápida mostrou que o primeiro carro foi inventado em 1885 então esse será nosso limite mínimo e o máximo será 2016 pois foi o último ano da coleta dos dados. </p>

In [20]:
autos = autos[autos['registration_year'].between(1885,2016)]
autos['registration_year'].describe()

count    356717.000000
mean       2002.781266
std           7.391978
min        1910.000000
25%        1999.000000
50%        2003.000000
75%        2008.000000
max        2016.000000
Name: registration_year, dtype: float64

## Analisando dados de marcas

In [21]:
# Mostra as marcas mais citadas
autos['brand'].value_counts()

volkswagen        75759
bmw               39131
opel              38198
mercedes_benz     34231
audi              31873
ford              24560
renault           16968
peugeot           10653
fiat               9189
seat               6647
skoda              5497
mazda              5475
smart              5032
citroen            4949
nissan             4841
toyota             4548
sonstige_autos     3786
hyundai            3507
mini               3286
volvo              3257
mitsubishi         2949
honda              2707
kia                2454
alfa_romeo         2264
suzuki             2253
porsche            2190
chevrolet          1789
chrysler           1406
dacia               874
daihatsu            780
jeep                780
land_rover          762
subaru              758
jaguar              614
trabant             577
saab                518
daewoo              513
rover               463
lancia              461
lada                218
Name: brand, dtype: int64

In [22]:
# Seleciona os 20 primeiros
autos['brand'].value_counts().index[:20]

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat', 'skoda', 'mazda', 'smart', 'citroen',
       'nissan', 'toyota', 'sonstige_autos', 'hyundai', 'mini', 'volvo'],
      dtype='object')

In [27]:
# Cria um dicionário para armazenar o for
top_20_marcas = {}

# Calcula o preço médio das marcas no top 20
for b in autos['brand'].value_counts().index[:20]:
    top_marcas = autos[autos['brand'] == b]
    preco_medio = top_marcas['price'].mean()
    top_20_marcas[b] = preco_medio

# Mostra o dicionário preenchido e suas chaves
for key in top_20_marcas:
    print(key, ":", top_20_marcas[key])

volkswagen : 5268.787101202497
bmw : 8315.099281899262
opel : 2896.157992565056
mercedes_benz : 8443.821243901726
audi : 8896.718194082767
ford : 3636.434609120521
renault : 2366.3506011315417
peugeot : 3206.2663099596357
fiat : 2803.857220589836
seat : 4420.369038664059
skoda : 6460.418046207022
mazda : 3970.6332420091326
smart : 3586.280802861685
citroen : 3649.0161648817943
nissan : 4585.39599256352
toyota : 5263.365875109938
sonstige_autos : 14302.470681458004
hyundai : 5487.679498146564
mini : 9979.38496652465
volvo : 5100.101627264354


In [28]:
# Cria um objeto de série
marca_valor = pd.Series(top_20_marcas)
print(marca_valor)

volkswagen         5268.787101
bmw                8315.099282
opel               2896.157993
mercedes_benz      8443.821244
audi               8896.718194
ford               3636.434609
renault            2366.350601
peugeot            3206.266310
fiat               2803.857221
seat               4420.369039
skoda              6460.418046
mazda              3970.633242
smart              3586.280803
citroen            3649.016165
nissan             4585.395993
toyota             5263.365875
sonstige_autos    14302.470681
hyundai            5487.679498
mini               9979.384967
volvo              5100.101627
dtype: float64


In [33]:
# Cria um dicionário para armazenar o for
km_top_20 = {}

# Calcula a média de kilometros para o top 20 de marcas
for b in autos['brand'].value_counts().index[:20]:
    top_brands = autos[autos['brand'] == b]
    media_km = top_brands['kilometer'].mean()
    km_top_20[b] = int(media_km)

# Nós mostra o resultado
for key in km_top_20:
    print(key, ":", km_top_20[key])

volkswagen : 128338
bmw : 132658
opel : 128754
mercedes_benz : 130579
audi : 129494
ford : 123617
renault : 127875
peugeot : 124599
fiat : 116523
seat : 120916
skoda : 113720
mazda : 125945
smart : 99684
citroen : 120313
nissan : 119543
toyota : 117337
sonstige_autos : 86724
hyundai : 104302
mini : 93566
volvo : 138101


In [34]:
# Cria um objeto de série para armazenar nosso cálculo de km por marca
marca_km = pd.Series(km_top_20)
print(marca_km)

volkswagen        128338
bmw               132658
opel              128754
mercedes_benz     130579
audi              129494
ford              123617
renault           127875
peugeot           124599
fiat              116523
seat              120916
skoda             113720
mazda             125945
smart              99684
citroen           120313
nissan            119543
toyota            117337
sonstige_autos     86724
hyundai           104302
mini               93566
volvo             138101
dtype: int64


In [39]:
# Cria um dataframe com o primeiro objeto de série que criamos
final_autos = pd.DataFrame(marca_valor, columns=['mean_price'])
final_autos

mean_price
volkswagen       5268.787101
bmw              8315.099282
opel             2896.157993
mercedes_benz    8443.821244
audi             8896.718194
ford             3636.434609
renault          2366.350601
peugeot          3206.266310
fiat             2803.857221
seat             4420.369039
skoda            6460.418046
mazda            3970.633242
smart            3586.280803
citroen          3649.016165
nissan           4585.395993
toyota           5263.365875
sonstige_autos  14302.470681
hyundai          5487.679498
mini             9979.384967
volvo            5100.101627

In [40]:
# Adiciona o segundo objeto de série como uma coluna
final_autos.insert(1, "mean_km", marca_km)
final_autos

mean_price  mean_km
volkswagen       5268.787101   128338
bmw              8315.099282   132658
opel             2896.157993   128754
mercedes_benz    8443.821244   130579
audi             8896.718194   129494
ford             3636.434609   123617
renault          2366.350601   127875
peugeot          3206.266310   124599
fiat             2803.857221   116523
seat             4420.369039   120916
skoda            6460.418046   113720
mazda            3970.633242   125945
smart            3586.280803    99684
citroen          3649.016165   120313
nissan           4585.395993   119543
toyota           5263.365875   117337
sonstige_autos  14302.470681    86724
hyundai          5487.679498   104302
mini             9979.384967    93566
volvo            5100.101627   138101

In [42]:
# Organizando a visualização do mais caro para o mais barato
final_autos.sort_values('mean_price', ascending=False)

mean_price  mean_km
sonstige_autos  14302.470681    86724
mini             9979.384967    93566
audi             8896.718194   129494
mercedes_benz    8443.821244   130579
bmw              8315.099282   132658
skoda            6460.418046   113720
hyundai          5487.679498   104302
volkswagen       5268.787101   128338
toyota           5263.365875   117337
volvo            5100.101627   138101
nissan           4585.395993   119543
seat             4420.369039   120916
mazda            3970.633242   125945
citroen          3649.016165   120313
ford             3636.434609   123617
smart            3586.280803    99684
peugeot          3206.266310   124599
opel             2896.157993   128754
fiat             2803.857221   116523
renault          2366.350601   127875

In [44]:
# Calcula a correlação entre as duas colunas
final_autos['mean_price'].corr(final_autos['mean_km'])

-0.4559256191477746

## Conclusões
<p> Havia muitos preenchimentos incorretos no dataset. Retiramos a coluna number_pictures porque estava vazia e retiramos da análise 14.811 linhas em decorrência das inconsistênciais encontradas na coluna price, resultando assim num dataset com 356717 linhas e 19 colunas. </p>
<p> Foi criada durante a análise um dataframe nomeado 'final_autos' para avaliar se havia correlação entre algumas colunas.A correlação entre marca, kilometragem e valor foi de apenas 45%, o que denota uma fraca correlação. </p>
<p> Dentre as 20 marcas analisadas, que foram escolhidas por serem as 20 mais citadas no dataset, a com a média de valor mais alta é a 'sonstige_autos' que apresenta a menor média de kilometragem. </p>